src/WireEncoding.chpl

```diff
@@ -57,211 +57,172 @@ module Encoding {
-    var a = fixed32Consume(ch);
+  proc floatConsumeBase(ch: readingChannel): real(32) throws {
+    var a = fixed32ConsumeBase(ch);
     var b: real(32);
     c_memcpy(c_ptrTo(b), c_ptrTo(a), c_sizeof(b.type));
     return b;
   }
 
-  proc doubleAppend(val: real(64), ch: writingChannel) throws {
+  proc doubleAppendBase(val: real(64), ch: writingChannel) throws {
     var a = val;
     var b: uint(64);
     c_memcpy(c_ptrTo(b), c_ptrTo(a), c_sizeof(b.type));
-    fixed64Append(b, ch);
+    fixed64AppendBase(b, ch);
   }
   
-  proc doubleConsume(ch: readingChannel): real(64) throws {
-    var a = fixed64Consume(ch);
+  proc doubleConsumeBase(ch: readingChannel): real(64) throws {
+    var a = fixed64ConsumeBase(ch);
     var b: real(64);
     c_memcpy(c_ptrTo(b), c_ptrTo(a), c_sizeof(b.type));
     return b;
   }
 
-  proc sfixed64Append(val: int(64), ch: writingChannel) throws {
+  proc sfixed64AppendBase(val: int(64), ch: writingChannel) throws {
     var a = val;
     var b: uint(64);
     c_memcpy(c_ptrTo(b), c_ptrTo(a), c_sizeof(b.type));
-    fixed64Append(b, ch);
+    fixed64AppendBase(b, ch);
   }
 
-  proc sfixed64Consume(ch: readingChannel): int(64) throws {
-    var a = fixed64Consume(ch);
+  proc sfixed64ConsumeBase(ch: readingChannel): int(64) throws {
+    var a = fixed64ConsumeBase(ch);
     var b: int(64);
     c_memcpy(c_ptrTo(b), c_ptrTo(a), c_sizeof(b.type));
     return b;
   }
 
-  proc sfixed32Append(val: int(32), ch: writingChannel) throws {
+  proc sfixed32AppendBase(val: int(32), ch: writingChannel) throws {
     var a = val;
     var b: uint(32);
     c_memcpy(c_ptrTo(b), c_ptrTo(a), c_sizeof(b.type));
-    fixed32Append(b, ch);
+    fixed32AppendBase(b, ch);
   }
 
-  proc sfixed32Consume(ch: readingChannel): int(32) throws {
-    var a = fixed32Consume(ch);
+  proc sfixed32ConsumeBase(ch: readingChannel): int(32) throws {
+    var a = fixed32ConsumeBase(ch);
     var b: int(32);
     c_memcpy(c_ptrTo(b), c_ptrTo(a), c_sizeof(b.type));
     return b;
   }
 
-  proc unknownField(fieldNumber, wireType, ch): bytes throws {
-    var s: bytes;
-    var tmpMem = openmem();
-    var memWriter = tmpMem.writer(kind=iokind.little, locking=false);
-    var memReader = tmpMem.reader(kind=iokind.little, locking=false);
-
-    tagAppend(fieldNumber, wireType, memWriter);
-    if wireType == varint {
-      var val = unsignedVarintConsume(ch)(0);
-      unsignedVarintAppend(val, memWriter);
-    } else if wireType == fixed64Type {
-      var val = fixed64Consume(ch);
-      fixed64Append(val, memWriter);
-    } else if wireType == lengthDelimited {
-      var val = bytesConsume(ch);
-      bytesAppend(val, memWriter);
-    } else if wireType == fixed32Type {
-      var val = fixed32Consume(ch);
-      fixed32Append(val, memWriter);
-    }
-
-    memWriter.close();
-    memReader.readbytes(s);
-    tmpMem.close();
-    return s;
-  }
-
   proc writeToOutputFileHelper(ref message, ch) throws {
```

Comment:
Should we keep these helper functions in this module only? I was thinking of making a separate `helper` or `utils` module.

Reply:
Sorry for the delay, I'm mostly hoping either @astatide or @mppf will review this instead of me.

In answer to your question, I'd keep helpers close to the code that uses them.  If you make a helper module, I'd want it to be a submodule and/or have a more unique name than `helper` or `utils`, so as not to cause issues with potential programs using the same names in their code.

Reply:
These helpers are used by the generated code. 

> If you make a helper module, I'd want it to be a submodule and/or have a more unique name than `helper` or `utils`, so as not to cause issues with potential programs using the same names in their code.

That makes sense. I will focus on this part when we will integrate the code to the main chapel repo next week :)